plugin/src/test/java/org/opensearch/ml/bwc/MLCommonsBackwardsCompatibilityIT.java

```diff
@@ -313,7 +319,17 @@ private String getModelIdWithFunctionName(FunctionName functionName) throws IOEx
     }
 
     private boolean isNewerVersion(String osVersion) {
-        return (Integer.parseInt(osVersion.substring(2, 3)) > 4) || (Integer.parseInt(osVersion.substring(0, 1)) > 2);
+        Pattern pattern = Pattern.compile("\\d+(?=\\.)");
+        Matcher matcher = pattern.matcher(osVersion);
+        ArrayList<Integer> osVersionArrayList = new ArrayList<>();
+        while (matcher.find()) {
+            osVersionArrayList.add(Integer.parseInt(matcher.group()));
+        }
+        if (osVersionArrayList.size() >= 2) {
+            return (osVersionArrayList.get(0) > 2 || osVersionArrayList.get(1) > 4);
```

Comment:
Thanks for fixing this. Can you add some comments for these magic numbers: 2, 4? Also add some example osVersion to explain ?

Reply:
Added.